app.js

```diff
@@ -27,5 +29,39 @@ app.get("/register", (req, res) => {
 app.get("/login", (req, res) => {
   res.render("login");
 });
+app.put("/:type", async(req, res, next) => {
+  if(req.params.type === "student")
+  {
+    const token = await jwtVerify(req.signedCookies.user);
+    const type = token.type;
+    if(token === false && type !== "student")
+    {
+      return res.sendStatus(401);
+    }
+    await updateStudentInfo(token.user, req.body.firstName, req.body.lastName, req.body.bio, req.body.gender);
+  }
+  else if(req.params.type === "parent")
+  {
+    const token = await jwtVerify(req.signedCookies.user);
+    const type = token.type;
+    if(token === false && type !== "parent")
+      return res.sendStatus(401);
+    await updateParentInfo(token.user, req.body.firstName, req.body.lastName);
+  }
+  else if(req.params.type === "tutor")
+  {
+    const token = await jwtVerify(req.signedCookies.user);
+    const type = token.type;
+    if(token === false && type !== "tutor")
+      return res.sendStatus(401);
+    await updateTutorInfo(token.user, req.body.firstName, req.body.lastName, req.body.bio, req.body.gender);
```

Comment:
This function can return `false` to denote that it failed. In that case, send in a `500`.

Reply:
9a4a464832b8a05a2ec4f1893f42397a3c4a522b